pkgs/development/tools/cambalache/default.nix

```diff
@@ -0,0 +1,95 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, python3
+, meson
+, ninja
+, pkg-config
+, gobject-introspection
+, desktop-file-utils
+, shared-mime-info
+, wrapGAppsHook
+, glib
+, gtk3
+, gtk4
+, webkitgtk
+, unstableGitUpdater
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "cambalache";
+  version = "unstable-2021-09-23";
+
+  format = "other";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "jpu";
+    repo = pname;
+    rev = "69b39dedbf9ccb3fac6afaa8f533d79e68d36792";
+    sha256 = "X+VOZQicUUrHsXm+ttLAPrtc94pvTdjDI8hIe/Zd3ks=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gobject-introspection # for setup hook
+    desktop-file-utils # for update-desktop-database
+    shared-mime-info # for update-mime-database
+    wrapGAppsHook
+  ];
+
+  pythonPath = with python3.pkgs; [
+    pygobject3
+    lxml
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
+    gtk4
+    webkitgtk
+  ];
+
+  # Not compatible with gobject-introspection setup hooks.
+  strictDeps = false;
+
+  # Prevent double wrapping.
+  dontWrapGApps = true;
+
+  postPatch = ''
+    patchShebangs postinstall.py
+  '';
+
+  preFixup = ''
+    # Let python wrapper use GNOME flags.
+    makeWrapperArgs+=(
+      # For gtk4-broadwayd
+      --prefix PATH : "${gtk4.dev}/bin"
```

Comment:
Should we use lib.getBin here?

Reply:
`lib.getBin` does not respect `outputBin` attribute.